maven-artifact/src/test/java/org/apache/maven/artifact/versioning/ComparableVersionTest.java

```diff
@@ -147,10 +154,10 @@ void testVersionsEqual() {
         checkVersionsEqual("1.0.0x", "1-x");
 
         // aliases
-        checkVersionsEqual("1ga", "1");
-        checkVersionsEqual("1release", "1");
-        checkVersionsEqual("1final", "1");
-        checkVersionsEqual("1cr", "1rc");
+        checkVersionsEqualOrder("1ga", "1");
```

Comment:
I'd be more comfortable if existing tests didn't change

Reply:
Because there was a previous case where `final` was replaced with an empty string and the hash comparison was different, I don't think this change will affect anything.

Reply:
still easier to follow if you add new tests without changing the existing test methods.

Reply:
I did not think of a particularly good way to ensure that the original test does not change, because the difference between checkVersionsEqualOrder and checkVersionsEqual is that the hash is no longer the same, let `1ga` be `1ga` itself, I think this is a good improvement. What do you think?

Reply:
Does checkVersionsEqual("1ga", "1") now fail? This is important.

If it does, then I need to dig into this much more carefully to understand what's going on.

If it doesn't, then the code shouldn't change here. It should be line by line identical.

New cases not previously tested should be in completely new test methods.

related: Existing utility methods should not change. New ones can (but probably shouldn't) be added. 

Assume the existing tests are correct unless you're deliberately changing behavior, and then call that out. However, do not assume the existing tests are well written examples of good testing practice that should be imitated. They aren't. The existing code makes a number of novice mistakes (testing more than one case in a single method; unclear method names, hiding test behavior in utility methods) that should not be copied in new code. There is no benefit to being consistent with bad practices.

Reply:
This code change will cause checkVersionsEqual("1ga", "1") to fail, and I think you can take a closer look at the specific changes.
The difference between `checkVersionsEqualOrder` and the `checkVersionsEqual` method is that `checkVersionsEqualOrder` only compares the order of versions, `checkVersionsEqual` compares Comparable HashCodes.
As for why I changed this test, I think I have explained above, because the previous logic was to directly change the contents of `1ga` to `1`, but now I will let `1ga` still stand for `1ga`, which will cause the hashcode in `checkVersionsEqual` to fail. This requires a new `checkVersionsEqualOrder` method to test that the order between `1ga` and `1` is consistent.